packages/radix-ui-themes/src/components/heading.css

```diff
@@ -9,9 +9,8 @@
   &:where([data-accent-color]) {
     color: var(--accent-a11);
 
-    &:where(.rt-high-contrast),
-    & .rt-Text:where(.rt-high-contrast) {
-      color: var(--accent-a12);
```

Comment:
Moved this to `text.css`. To avoid unexpected style overrides, our general rule that when nesting components has an effect on styles, we should either use the CSS custom properties cascade, or style the final targeted node in its own file.

For example, if Heading changes what Text looks like, the rule should be in `text.css`, not in `heading.css`. Alternatively, we could define a variable for Text that Heading can override via the variabel cascade.